js/shopping-list.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+class ShoppingList {
+  constructor() {
+    this.items = [];
+  }
+
+  addItem(item) {
+    if (!(item instanceof ShoppingListItem)) {
+      throw new Error('invalid item');
+    }
+    this.items.push(item);
+  }
+
+  removeItem(item) {
+    if (item === undefined) {
+      this.items.pop();
+    } else if (
+      !(item instanceof ShoppingListItem && this.items.includes(item))
```

Comment:
This would mean that it's both not a shopping list item, and not in the array. It's probably best to split this out into separate cases and handle them accordingly.